app/uk/gov/hmrc/transitmovementsrouter/connectors/PushNotificationsConnector.scala

```diff
@@ -61,7 +62,7 @@ class PushNotificationsConnectorImpl @Inject() (httpClientV2: HttpClientV2, appC
   private def pushNotificationMessageUpdate(movementId: MovementId, messageId: MessageId): UrlPath =
     UrlPath.parse(s"$baseRoute/traders/movements/${movementId.value}/messages/${messageId.value}")
 
-  override def post(movementId: MovementId, messageId: MessageId, source: Option[Source[ByteString, _]])(implicit
+  override def post(movementId: MovementId, messageId: MessageId, sourceXml: Option[Source[ByteString, _]], sourceJson: Option[JsValue])(implicit
```

Comment:
I do not like this at all. We've now got one method that does the following:

* Send an XML payload
* Send a Json payload
* Send no payload

An optional was okay before, but it's not really okay now when we have more complicated situations.

There should now be three methods, one for each scenario. There can be a common private method that they feed in to, no problem, but this is just a messy signature - just have overloaded methods.